content/en/blog/_posts/2023-12-07-a-demo-of-webassembly-and-wasi-on-kubernetes/index.md

```diff
@@ -0,0 +1,864 @@
+			"options": [
+				"nosuid",
+				"noexec",
+				"nodev",
+				"relatime",
+				"ro"
+			]
+		}
+	],
+	"annotations": {
+		"module.wasm.image/variant": "compat"
+	},
+	"linux": {
+		"resources": {
+			"devices": [
+				{
+					"allow": false,
+					"access": "rwm"
+				}
+			]
+		},
+		"namespaces": [
+			{
+				"type": "pid"
+			},
+			{
+				"type": "network",
+				"path": "/proc/1/ns/net"
+			},
+			{
+				"type": "ipc"
+			},
+			{
+				"type": "uts"
+			},
+			{
+				"type": "cgroup"
+			},
+			{
+				"type": "mount"
+			}
+		],
+		"maskedPaths": [
+			"/proc/acpi",
+			"/proc/asound",
+			"/proc/kcore",
+			"/proc/keys",
+			"/proc/latency_stats",
+			"/proc/timer_list",
+			"/proc/timer_stats",
+			"/proc/sched_debug",
+			"/sys/firmware",
+			"/proc/scsi"
+		],
+		"readonlyPaths": [
+			"/proc/bus",
+			"/proc/fs",
+			"/proc/irq",
+			"/proc/sys",
+			"/proc/sysrq-trigger"
+		]
+	}
+}
+```
+
+Start the container using crun. 
+
+```shell
+crun run wasm-demo-app
+```
+
+Send a request to the service.
+
+```bash
+curl http://localhost:8080
+```
+
+The output is:
+
+```bash
+Hello, World!
+```
+
+To delete the container, you can execute the following command:
+
+```bash
+crun kill wasm-demo-app SIGKILL
+```
+
+#### High-level Container Runtime
+
+The _container shim_ serves as a bridge between high and low-level container runtimes. Its main purpose is to abstract low-level runtime details, enabling uniform management of various runtimes in high-level runtime. In this section, I will use _[containerd](https://github.com/containerd/containerd)_ as an example. Containerd is an industry-standard container runtime with an emphasis on simplicity, robustness, and portability. 
+
+Containerd can manage Wasm modules in two ways:
+- 1.Manages Wasm modules through container runtimes like crun and youki that support building with the Wasm runtime library. These two runtimes can also run regular Linux containers. Containerd uses `containerd-shim-runc-v2` to interface with low-level container runtimes.
+- 2.Manages Wasm modules directly through Wasm runtimes, such as WasmEdge, Wasmtime, Spin, and Slight. Containerd uses `containerd-wasm-shim(runwasi)` to interface with Wasm runtimes.
+
+{{< figure src="03-use-containerd-shim-to-manage-wasm-modules.svg" alt="The diagram illustrates that containerd can manage WebAssembly modules in two ways: through container runtimes that support building with the WebAssembly runtime library via the containerd-shim-runc-v2, or directly through WebAssembly runtimes like WasmEdge and Wasmtime using the containerd-wasm-shim." caption="Figure 3: Use containerd shim to manage Wasm modules" >}}
+
+##### Containerd + Crun
+
+In this section, I will demonstrate how to configure crun as runtime in containerd, enabling support for running Wasm modules.
+
+{{< note >}}
+Ensure that crun binary has been installed as per the instructions in the section: [Low-level Container Runtime](#low-level-container-runtime).
```

Comment:
You might need to update the link here if following feedback elsewhere.

Reply:
Updated the link: https://github.com/kubernetes/website/pull/43760/commits/d4ee5b15713b53f4baa41c365d2a36d6533f4aa3